### PR TITLE
Barcode scanning functionality working.

### PR DESCRIPTION
**Pull request by hpabst**
A user can create a listing by scanning a book's barcode. Scanning the barcode will automatically insert information about the book into the listing creation screen.

### CODE REVIEW

File: TextCrunch/BarcodeScanViewController.swift [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    var highlightView : UIView = UIView()
+    var isbnString: String = ""
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Allow the view to resize freely
+        self.highlightView.autoresizingMask =   UIViewAutoresizing.FlexibleTopMargin |
+            UIViewAutoresizing.FlexibleBottomMargin |
+            UIViewAutoresizing.FlexibleLeftMargin |
```
Comment (gcoomber) [inline TextCrunch/BarcodeScanViewController.swift]: Out of interest, what's this autoresizing stuff, is it for supporting other devices?

File: TextCrunch/BarcodeScanViewController.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    let session : AVCaptureSession = AVCaptureSession()
+    var previewLayer: AVCaptureVideoPreviewLayer!
+    var highlightView : UIView = UIView()
+    var isbnString: String = ""
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Allow the view to resize freely
+        self.highlightView.autoresizingMask =   UIViewAutoresizing.FlexibleTopMargin |
```
Comment (gcoomber) [inline TextCrunch/BarcodeScanViewController.swift]: The double spaces throughout this function can be removed.

File: TextCrunch/BarcodeScanViewController.swift [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        
+
+        // Start the scanner. You'll have to end it yourself later.
+        session.startRunning()
+        
+    }
+    
+    // This is called when we find a known barcode type with the camera.
+    func captureOutput(captureOutput: AVCaptureOutput!, didOutputMetadataObjects metadataObjects: [AnyObject]!, fromConnection connection: AVCaptureConnection!) {
+
```
Comment (gcoomber) [inline TextCrunch/BarcodeScanViewController.swift]: can get rid of extra empty space and have these three vars together in three lines.

File: TextCrunch/BarcodeScanViewController.swift [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            AVMetadataObjectTypeEAN13Code,
+            AVMetadataObjectTypeEAN8Code,
+            AVMetadataObjectTypeCode93Code,
+            AVMetadataObjectTypeCode128Code,
+            AVMetadataObjectTypePDF417Code,
+            AVMetadataObjectTypeQRCode,
+            AVMetadataObjectTypeAztecCode
+        ]
+        
+
```
Comment (gcoomber) [inline TextCrunch/BarcodeScanViewController.swift]: extra spaces throughout function can be removed.

**Comment by gcoomber** [comment]
So the UPC scanner is working with this change and the user can list info after the scan?

File: TextCrunch/BarcodeScanViewController.swift (hunk 1 above)
Comment (hpabst) [inline TextCrunch/BarcodeScanViewController.swift]: The highlight view is a small line that is drawn over top of the barcode when it is successfully scanned. This autoresizing stuff allows for the line to be drawn no matter the angle or distance the camera is at when it is registered.

**Pull request merged**